Pinta.Core/Effects/HsvColor.cs

```diff
@@ -6,141 +6,66 @@
 /////////////////////////////////////////////////////////////////////////////////
 
 using System;
+using Cairo;
+using Pinta.Core.Extensions;
 
 namespace Pinta.Core;
 
 /// <summary>
-/// Adapted from: 
+/// Adapted from:
 /// "A Primer on Building a Color Picker User Control with GDI+ in Visual Basic .NET or C#"
 /// http://www.msdnaa.net/Resources/display.aspx?ResID=2460
 /// </summary>
 public readonly struct HsvColor
 {
-	public readonly int Hue { get; } // 0-360
-	public readonly int Saturation { get; } // 0-100
-	public readonly int Value { get; } // 0-100
+	public double Hue { get; init; } // 0-360
+	public double Sat { get; init; } // 0-1
+	public double Val { get; init; } // 0-1
 
-	public static bool operator == (HsvColor lhs, HsvColor rhs)
-		=> lhs.Hue == rhs.Hue && lhs.Saturation == rhs.Saturation && lhs.Value == rhs.Value;
-
-	public static bool operator != (HsvColor lhs, HsvColor rhs)
-		=> !(lhs == rhs);
-
-	public override readonly bool Equals (object? obj)
-		=> obj is HsvColor hsv && this == hsv;
+	public static bool operator == (HsvColor c1, HsvColor c2)
```

Comment:
There are some compiler warnings related to this:
```
'HsvColor' defines operator == or operator != but does not override Object.Equals(object o)
```

Using a fixed tolerance inside operator== makes me feel a bit uneasy - if this is just for unit test purposes, maybe just make this into a separate method like `AlmostEquals()` or something like that?